hs_app_netCDF/models.py

```diff
@@ -236,72 +405,10 @@ def get_xml(self, pretty_print=True):
 
         # inject netcdf resource specific metadata element 'variable' to container element
         for variable in self.variables.all():
-            md_fields = {
-                "md_element": "netcdfVariable",
-                "name": "name",
-                "unit": "unit",
-                "type": "type",
-                "shape": "shape",
-                "descriptive_name": "longName",
-                "method": "comment",
-                "missing_value": "missingValue"
-            }  # element name : name in xml
-            self.add_metadata_element_to_xml(container, variable, md_fields)
-
-        if self.ori_coverage.all().first():
-            ori_cov_obj = self.ori_coverage.all().first()
-            hsterms_ori_cov = etree.SubElement(container, '{%s}spatialReference' %
-                                               self.NAMESPACES['hsterms'])
-            cov_term = '{%s}' + 'box'
-            hsterms_coverage_terms = etree.SubElement(hsterms_ori_cov, cov_term %
-                                                      self.NAMESPACES['hsterms'])
-
-            hsterms_ori_cov_rdf_Description = etree.SubElement(hsterms_coverage_terms, '{%s}value' %
-                                                               self.NAMESPACES['rdf'])
-            cov_box = ''
-
-            # add extent info
-            if ori_cov_obj.value:
-                cov_box = 'northlimit=%s; eastlimit=%s; southlimit=%s; westlimit=%s; unit=%s' \
-                        % (ori_cov_obj.value['northlimit'], ori_cov_obj.value['eastlimit'],
-                           ori_cov_obj.value['southlimit'], ori_cov_obj.value['westlimit'],
-                           ori_cov_obj.value['units'])
-
-            if ori_cov_obj.value.get('projection'):
-                cov_box += '; projection_name={}'.format(ori_cov_obj.value['projection'])
-
-            if ori_cov_obj.projection_string_text:
-                cov_box += '; projection_string={}'.format(ori_cov_obj.projection_string_text)
-
-            if ori_cov_obj.datum:
-                cov_box += '; datum={}'.format(ori_cov_obj.datum)
-
-            hsterms_ori_cov_rdf_Description.text = cov_box
-
-        return etree.tostring(RDF_ROOT, pretty_print=pretty_print)
-
-    def add_metadata_element_to_xml(self, root, md_element, md_fields):
-        from lxml import etree
-        element_name = md_fields.get('md_element') if md_fields.get('md_element') \
-            else md_element.term
+            variable.add_to_xml_container(container)
 
-        hsterms_newElem = etree.SubElement(
-            root,
-            "{{{ns}}}{new_element}".format(ns=self.NAMESPACES['hsterms'], new_element=element_name))
+        ori_cov_obj = self.ori_coverage.all().first()
```

Comment:
There is a reason why GenericRelation has been used - same object type can be related to different types (in this case NetCDF resource type and NetCDFLogicalFile type) . The OriginalCoverage class has the constraint that only one instance of it can be related to only one instance of some other type. That means ```self.orig_coverage.all()``` also will only give one value at most. There is also a history to why we had to use GenericRelation for metadata elements. Before proxy classes for various resource types we had originally all concrete classes for each of the resource types. Once we moved to resource proxy classes we thought of refactoring the metadata implementation not to use GenericRelation. In fact there is a GitHub issue for that. But that issue never got prioritized probably because the existing implementation has been working. When we move to key/value all these will change for better I think.